hw/ip/dma/doc/programmers_guide.md

```diff
@@ -0,0 +1,93 @@
+
+## Initiate a Memory transfer
+
+To start a memory transfer, software needs to configure several registers that define the source and destination of the data, as well as the transfer size and access pattern:
+
+1.  **Source and Destination Configuration:** Configure the source and destination address modes using the [`SRC_CONFIG`](registers.md#src_config) and [`DST_CONFIG`](registers.md#dst_config) registers.
+2.  **Source and Destination Addresses:** Specify the starting memory addresses for the source and destination using the lower and upper 32-bit parts of the addresses in the [`SRC_ADDR_LO`](registers.md#src_addr_lo), [`SRC_ADDR_HI`](registers.md#src_addr_hi), [`DST_ADDR_LO`](registers.md#dst_addr_lo), and [`DST_ADDR_HI`](registers.md#dst_addr_hi) registers.
+3.  **Total Transfer Size:** Define the total number of bytes to be transferred using the [`TOTAL_DATA_SIZE`](registers.md#total_data_size) register.
+4.  **Access Stride:** Configure the access stride (the number of bytes accessed in each burst) using the [`TRANSFER_WIDTH`](registers.md#transfer_width) register.
+5.  **Chunk Size (Memory-to-Memory):** The DMA performs memory access in chunks. For a standard memory-to-memory transfer, the [`CHUNK_DATA_SIZE`](registers.md#chunk_data_size) register is typically set to the same value as the [`TOTAL_DATA_SIZE`](registers.md#total_data_size) register. The concept of chunked transfers is further explained in the [Chunked Data Transfers](#Chunked_Data_Transfers) section.
+6.  **Start the Transfer:** Initiate the transfer by writing to the `go` bit in the [`CONTROL`](registers.md#control) register.
+7.  **Monitor Transfer Completion:** After starting the transfer, software can monitor its progress by either polling the [`STATUS`](registers.md#status) register or by waiting for a specific interrupt to be raised.
+
+### Interrupt Handling
+
+The DMA can signal various events to the software through interrupts.
+The following three types of interrupts are supported:
+
+1.  **Transfer Completion:** An interrupt is raised when the entire data transfer (as defined by `TOTAL_DATA_SIZE`) is complete.
+2.  **Chunk Completion:** An interrupt is raised after the transfer of a single chunk of data (as defined by `CHUNK_DATA_SIZE`) is finished.
+3.  **Error Condition:** An interrupt is raised if an error occurs during the transfer process.
+
+The current status of the DMA, including pending interrupts, can be read from the [`STATUS`](registers.md#status) register.
+Since interrupts are implemented as status bits, they are cleared by writing a '1' to the corresponding bit in the `STATUS` register.
+
+### Aborting a Transfer
+
+Software can terminate an ongoing DMA transfer at any point by writing to the `abort` bit in the [`CONTROL`](registers.md#control) register.
```

Comment:
Worth mentioning that the status register indicates that the abort has completed and that it does so almost immediately; no need to schedule any timer/await an interrupt.